pkg/hub/submarineragent/controller.go

```diff
@@ -371,23 +375,27 @@ func (c *submarinerAgentController) syncSubmarinerConfig(ctx context.Context,
 
 	// prepare submariner cluster environment
 	errs := []error{}
-	cloudProvider, preparedErr := cloud.GetCloudProvider(c.kubeClient, c.manifestWorkClient, c.eventRecorder, managedClusterInfo, config)
-	if preparedErr == nil {
-		preparedErr = cloudProvider.PrepareSubmarinerClusterEnv()
-	}
+	var condition *metav1.Condition
+	if managedClusterInfo.Platform != "GCP" {
```

Comment:
Instead of falling thru as a no-op and modifying `UpdateSubmarinerConfigStatusFn` to handle nil condition, just return here:

```suggestion
	if managedClusterInfo.Platform == "GCP" {
	    return nil
	}
```

Reply:
it is not a no-op , helpers.UpdateSubmarinerConfigStatusFn(condition, managedClusterInfo), the managedclusterInfo is updated and is required in spoke

Reply:
oh right.